step-release-vis/src/app/services/environment_test.ts

```diff
@@ -13,7 +14,23 @@ describe('EnvironmentService', () => {
     service = TestBed.inject(EnvironmentService);
   });
 
-  it('should be created', () => {
-    expect(service).toBeTruthy();
+  // TODO(naoai): write getPolygons test
+  it('#getPercentages should return 3 candidates with ~ equal percentages', () => {
+    const input: CandidateInfo[] = [{name:  '1', job_count: 666}, {name: '2', job_count: 667},
+      {name: '3', job_count: 667}];
+    // @ts-ignore
+    const resultMap: Map<string, number> = service.getPercentages(input);
+
+    expect(resultMap.get('1')).toEqual(33.3);
+    expect(resultMap.get('2')).toEqual(33.35);
+    expect(resultMap.get('3')).toEqual(33.35);
+  });
+
+  it('#getPercentages should have a candidate with 100', () => {
```

Comment:
What if your CandidateInfo list is empty for some reason? What if the job_count is 0 for every candidate?
(These could be some error cases which should be tested / handled in the code)

Reply:
what would a feasible output look like if the job_count sum is 0? that should mean there is an error with the data if I understand correctly.

Reply:
should I throw an error message?

Reply:
@stofee Do we need to handle this kind of errors here?

Reply:
ok, lets say we can trust the input will have job_count more than 0, but the empty list should be tested.